### PR TITLE
Add highlight alert to be able to disable highlighting

### PR DESCRIPTION
**Pull request by thektan**
For #155 

I ended up styling it like an alert so it wouldn't mix with the navigation links. It'll also truncate on longer search terms. Let me know your thoughts.

Screen Shot 2021-05-04 at 11 15 11 PM
Truncated:
Screen Shot 2021-05-04 at 11 14 55 PM

---

Here's some other options I briefly explored:
The standard link seemed it could be confusing mixing JS actions and navigation links.
Screen Shot 2021-05-04 at 2 36 14 PM
Screen Shot 2021-05-04 at 1 23 27 PM

Top page alert blocks more content and if the help center alert isn't closed yet, the stacked alerts would make the page look really busy. 
Screen Shot 2021-05-04 at 8 53 12 PM

cc @rdai10 @rbohl @lipusz @jrhoun

### CODE REVIEW

**Comment by lipusz** [comment]
Thank you Kevin! 👏

**Comment by lipusz** [comment]
https://issues.liferay.com/browse/LRDOCS-9397

File: site/docs/_static/js/highlighting-alert.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const HIGHLIGHT_TEXT_MATCH_ID = 'highlightTextMatch';
+const REMOVE_HIGHLIGHT_LINK_ID = 'removeHighlightLink';
+
+const HIGHLIGHT_PARAM = 'highlight';
+
+const TRUNCATE_LENGTH = 50;
+
+/**
+ * Calls Sphinx utility to remove highlighted text.
+ * https://github.com/sphinx-doc/sphinx/blob/9e1b4a8f1678e26670d34765e74edf3a3be3c62c/sphinx/themes/basic/static/doctools.js#L261-L267
```
Comment (rdai10) [inline site/docs/_static/js/highlighting-alert.js:16]: Nice!

**Review by rdai10 — CHANGES_REQUESTED** [review]
Thanks @thektan , very comprehensive! The code looks good, the only comment I have is styling the highlighting term as admonition. Admonition have special meaning in the flow of the text, mixing it in the nav dilutes its association. A suggestion might be to keep the background color and alert icon but remove the left border so the message still stands out but is different enough from an admonition. 

I also thought calling it "Search Matches" in your third screenshot may be more obvious to the average user, since learn's audience is broad. What do you think?

@jrhoun , any feedback?

**Comment by jrhoun** [comment]
@thektan - 

Thanks for the pull. RE:

> The code looks good, the only comment I have is styling the highlighting term as admonition. Admonition have special meaning in the flow of the text, mixing it in the nav dilutes its association. A suggestion might be to keep the background color and alert icon but remove the left border so the message still stands out but is different enough from an admonition.

I would agree with this if it is possible. As far as the terms shown, I think `Highlighting "xyz"` is fine to me.

**Comment by thektan** [comment]
Updated! 
Screen Shot 2021-05-10 at 10 57 30 AM

**Comment by jrhoun** [comment]
Merged. Thanks a lot!

**Pull request closed without merging**